[PoC] feat: Clean cache command (#1394)
SAP/ui5-tooling: internal/documentation/docs/pages/Troubleshooting.md

## UI5 Project
### `~/.ui5` Taking too Much Disk Space

> There are possibly many versions of UI5 framework dependencies installed on your system, taking a large amount of disk space.

Reviewer: Since starting this PR, I have enhanced this section (see https://github.com/UI5/cli/pull/1427). Is it intended to remove the information again?

Author: I have reverted the Troubleshooting doc to the one in the main branch and have adjusted to address the cache clean solution instead of directory removal

#### Resolution

Use the dedicated cache clean command, which safely removes all cached data:

```sh
ui5 cache clean
```

This will display the cache location, the amount of data that will be removed, and ask for confirmation before proceeding. To skip the confirmation prompt (e.g. in CI environments), use the `--yes` flag:

```sh
ui5 cache clean --yes
```

The command removes the following cached data:
- **UI5 Framework packages** — downloaded UI5 library files (`~/.ui5/framework/`)
- **Build cache (DB)** — build data (`~/.ui5/buildCache/`)
- **Orphaned framework data** — incomplete framework directories left over from previously interrupted clean operations (`~/.ui5/.framework_to_delete_*/`)

Any missing framework dependencies will be downloaded again during the next UI5 CLI invocation.

::: info

> If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5 config set ui5DataDir`, replace `~/.ui5/` with that path. See [Changing UI5 CLI's Data Directory](#changing-ui5-cli-s-data-directory).

Reviewer: This info is not applicable anymore, or is it? It was referring to the paths for `rm -rf`. I think it needs to be updated to fit into the context of the new command.

Author: I still see a value in that i.e. `UI5_DATA_DIR=custom/path ui5 cache clean -y`
Maybe, I can reword it.

What do you think?

Reviewer: Yes, I think so too. But where does

> replace `~/.ui5/` with that path.

apply here now?

Author: If you take a look several lines above where the CLI output is explained, there are the full paths to the cache dirs that include `~/.ui5/...`. From that perspective, my expectation here would be  to explain that this is the dafault UI5 CLI home directory and you need to be aware that if you have replaced it with something else via `UI5_DATA_DIR` or `ui5 config set ui5DataDir` during the build, for example, you must also use that for the cache cleanup.

I will reword this into that direction

Author: What do you think about this rewording: https://github.com/UI5/cli/pull/1394/commits/3f5c3d367fb594ff41e388558900ab33d22112cc

Reviewer: Thats great!

#### Resolution

Use the dedicated cache clean command, which safely removes all cached data:

```sh
ui5 cache clean
```

This will display the cache location, the amount of data that will be removed, and ask for confirmation before proceeding. To skip the confirmation prompt (e.g. in CI environments), use the `--yes` flag:

```sh
ui5 cache clean --yes
```

The command removes two types of cached data:
- **UI5 Framework packages** — downloaded UI5 library files (`~/.ui5/framework/`)
- **Build cache (DB)** — build data (`~/.ui5/buildCache/`)

Any missing framework dependencies will be downloaded again during the next UI5 CLI invocation.

::: info

> If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5DataDir`, the cache will be cleaned from that location instead of `~/.ui5`. See [Changing UI5 CLI's Data Directory](#changing-ui5-clis-data-directory) below.

Reviewer: Previous content had `ui5 config set ui5DataDir` instead of just `ui5DataDir`.

The anchor link was correct before (https://ui5.github.io/cli/v5/pages/Troubleshooting#changing-ui5-cli-s-data-directory).

Author: Reverted with https://github.com/UI5/cli/pull/1394/commits/4778dbb25ed9c375e8ee8bcede527be062a0ee76

#### Resolution

Use the dedicated cache clean command, which safely removes all cached data:

```sh
ui5 cache clean
```

> This will display the cache location, the amount of data that will be removed, and ask for confirmation before proceeding. To skip the confirmation prompt (e.g. in CI environments), use the `--yes` flag:

Reviewer 2: ```suggestion
This displays the cache location, the amount of data that gets removed, and asks for confirmation before proceeding. To skip the confirmation prompt (for example in CI environments), use the `--yes` flag:
```

```sh
ui5 cache clean --yes
```

The command removes the following cached data:

> - **UI5 Framework packages** — downloaded UI5 library files (`~/.ui5/framework/`)

Reviewer 2: ```suggestion
- **UI5 framework packages** — downloaded UI5 library files (`~/.ui5/framework/`)
```

- **Build cache (DB)** — build data (`~/.ui5/buildCache/`)

> - **Orphaned framework data** — incomplete framework directories left over from previously interrupted clean operations (`~/.ui5/.framework_to_delete_*/`)

Reviewer 2: ```suggestion
- **Orphaned framework data** — incomplete framework directories left over from previously interrupted cleanup operations (`~/.ui5/.framework_to_delete_*/`)
```

> Any missing framework dependencies are downloaded during the next UI5 CLI invocation.

Reviewer 3: ```suggestion
Any required framework dependencies will be re-downloaded during the next UI5 CLI invocation.
```

> Any missing framework dependencies will be downloaded again during the next UI5 CLI invocation.

Reviewer 2: ```suggestion
Any missing framework dependencies are downloaded during the next UI5 CLI invocation.
```

::: info

> If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5 config set ui5DataDir`, the `ui5 cache clean` command automatically cleans up that location instead of the default `~/.ui5/`. See [Changing UI5 CLI's Data Directory](#changing-ui5-cli-s-data-directory).

Reviewer 3: ```suggestion
If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5 config set ui5DataDir`, the `ui5 cache clean` command will clean up that location instead of the default `~/.ui5/`. See [Changing UI5 CLI's Data Directory](#changing-ui5-cli-s-data-directory).
```

> If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5 config set ui5DataDir`, the `ui5 cache clean` command automatically cleans that location instead of the default `~/.ui5/`. See [Changing UI5 CLI's Data Directory](#changing-ui5-cli-s-data-directory).

Reviewer 2: ```suggestion
If you have configured a custom data directory via `UI5_DATA_DIR` or `ui5 config set ui5DataDir`, the `ui5 cache clean` command automatically cleans up that location instead of the default `~/.ui5/`. See [Changing UI5 CLI's Data Directory](#changing-ui5-cli-s-data-directory).
```
